packages/gatsby-plugin-functions/src/gatsby-node.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "fs-extra"
+import glob from "globby"
+import path from "path"
+import webpack from "webpack"
+
+import { ParentSpanPluginArgs, PluginOptions } from "gatsby"
+
+export async function onPreInit(
+  { reporter }: ParentSpanPluginArgs,
+  { path: functionsDirectoryPath }: PluginOptions
+): Promise<void> {
+  const activity = reporter.activityTimer(`building functions`)
+  activity.start()
+
+  const functionsDirectory = path.resolve(
+    process.cwd(),
+    functionsDirectoryPath as string
```

Comment:
Rather than coercing, you should check that `functionsDirectoryPath` is set and exists